text/0000-scriptless-buildpacks.md

```diff
@@ -0,0 +1,204 @@
+name = "Warbler Buildpack"
+version = "0.0.9"
+
+[buildpack.detect]
+requires = [
+  "jruby",
+  "warbler",
+  "bundler"
+]
+provides = [
+  "war"
+]
+
+# if there is a bin/build, this will always run first.
+[buildpack.build]
+run = [
+  "rake war",
+  "rake db:migrate"
+]
+
+[[buildpack.build.launch.processes]]
+type = "web"
+command = "java -jar myapp.war"
+```
+
+The buildpack requires `jruby`, `warbler`, and `bundler` be installed by previous buildpacks. It then provides a `war` file.
+
+The build will run two `rake` commands, and then it generates a default `web` command. But no layers are created.
+
+## Example: JDK buildpack
+
+This buildpack is functionally similar to the [sample Java buildpack](https://github.com/buildpack/samples/blob/master/java-buildpack/).
+
+```toml
+
+[buildpack]
+id = "io.buildpacks.zulu"
+name = "Azul Zulu Buildpack"
+version = "0.0.9"
+
+[buildpack.detect]
+provides = [
+  "jdk"
+]
+
+[[buildpack.build.layers]]
+id = "jdk"
+cache = "true"
+launch = "true"
+build = "true"
+run = [
+  "export JDK_URL=https://cdn.azul.com/zulu/bin/zulu8.28.0.1-jdk8.0.163-linux_x64.tar.gz",
+  "wget -q -O - $JDK_URL | tar pxz -C $1/jdk --strip-components=1"
+]
+
+  [buildpack.build.layers.metadata]
+  version = "zulu-1.8.0_163"
+
+  [buildpack.build.layers.env]
+  JAVA_HOME = "$1/jdk"
+  LD_LIBRARY_PATH = "$JAVA_HOME/jre/lib/amd64/server"
+
+  [[buildpack.build.layers.profile]]
+  name = "jdk.sh"
+  script = [
+    "export JAVA_HOME=$1/jdk",
+    "export LD_LIBRARY_PATH=$JAVA_HOME/jre/lib/amd64/server"
+  ]
+
+[[buildpack.build.launch.processes]]
+type = "web"
+command = "java -jar *.jar"
+```
+
+The buildpack provides a `jdk`. It has one layer that is made available to launch, build, and cache.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- We might be encouraging a bad practice. While the scriptless buildpack is extremely useful, it may drive developers to code themselves into a corner when the buildpack eventually becomes complex enough that it needs to be rewritten as a buildpack with scripts.
+- The logic around regenerating a layer or not may be confusing and ultimately intractable.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- The inline buildpack approach described in the Project Descriptor RFC helps solve this problem too, but still requires the scripts.
+- Instead of tables under `buildpack`, we could use top-level `[detect]` and `[build]` tables. This matches the top-level `[[order]]` and `[[stacks]]` tables, but might cause confusion because these represent very different concepts.
+
+# Prior Art
+[prior-art]: #prior-art
+
+- `Dockerfile` is similar in the use case it addresses.
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- How do we merge the scriptless buildpack with scripts in `bin/` if they are eventually created?
+- What is the migrate path for a developer who has created a fairly complex scriptless buildpack, and now wants to port it to a `bin/build`.
```

Comment:
It would be cool to just have a way to `generate` the `bin/build`/`bin/detect` binaries from this setup.